docs/DeveloperGuide.adoc

```diff
@@ -153,98 +173,236 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+Activity diagram:
 
-image::UndoRedoState3.png[]
+image::AddDueSoonActivityDiagram.png[width = "200", length = "300"]
+
+Step 2. The user executes `add` command to add a task.
+The `add` command execution checks if this task belongs in the due soon tasks list (i.e it is due in 7 days). If it does, the task is added to both lists, else it is only added to the main list.
+
+Step 3. The user executes `delete` command to delete a task.
+The `delete` command execution also checks if this task exists in the due soon task list. If it does, task is deleted from both lists, else it is only deleted from the main list.
+
+Step 4. The user executes `due soon` command to refresh the due soon task list.
+The `due soon` command execution will traverse the main list and check if there are any inconsistencies. For example, if there has been a change in time and a task is missing from the due soon task list, it will be added.
 
 [NOTE]
-If the `currentStatePointer` is at index 0, pointing to the initial address book state, then there are no previous address book states to restore. The `undo` command uses `Model#canUndoAddressBook()` to check if this is the case. If so, it will return an error to the user rather than attempting to perform the undo.
+If there are no such inconsistencies, the due soon command will not make any changes.
+
+Activity diagram:
 
-The following sequence diagram shows how the undo operation works:
+image::DueSoonActivityDiagram.png[width = "400", length = "600"]
 
-image::UndoSequenceDiagram.png[]
+Sequence diagram:
 
-NOTE: The lifeline for `UndoCommand` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
+image::DueSoonSequenceDiagram.png[width = "600", length = "500"]
 
-The `redo` command does the opposite -- it calls `Model#redoAddressBook()`, which shifts the `currentStatePointer` once to the right, pointing to the previously undone state, and restores the address book to that state.
+==== Design Considerations
+
+===== Aspect: How `due soon` command works
+
+* **Alternative 1 (current choice):** Does not clear and reload the due soon task list.
+** Pros: Fewer operations need to be performed.
+** Cons: May have be more prone to errors because list is not completely "refreshed". Some parts of it are the same.
+* **Alternative 2:** Clear and reload the due soon task list.
+** Pros: Less prone to errors since the entire list is "refreshed".
+** Cons: More operation intensive.
+// end::dueSoon[]
+
+// tag::taskSummary[]
+=== Task Summary Feature
+
+==== Implementation
+
+The task summary feature allows user to enjoy visualized and real-time statistics of their tasks' information.
+
+This feature is implemented using different charts as `pie chart`, `area chart`, `bar chart` and `line chart`.
```

Comment:
I'm not sure how these charts are implemented, are they new classes or just some attributes? maybe for OOP, implement them as new classes and reflect that in model?